BUILD.bazel

```diff
@@ -316,3 +340,140 @@ pyx_library(
+        "Language.py",
+        "ObjectTableData.py",
+        "ProfileEvent.py",
+        "ProfileTableData.py",
+        "RayResource.py",
+        "ResourcePair.py",
+        "SchedulingState.py",
+        "TablePrefix.py",
+        "TablePubsub.py",
+        "TaskInfo.py",
+        "TaskLeaseData.py",
+        "TaskReconstructionData.py",
+        "TaskTableData.py",
+        "TaskTableTestAndUpdate.py",
+    ],
+    out_prefix = "python/ray/core/generated/",
+)
+
+flatbuffer_py_library(
+    name = "python_node_manager_fbs",
+    srcs = [
+        "src/ray/raylet/format/node_manager.fbs"
+    ],
+    outs = [
+        "DisconnectClient.py",
+        "FetchOrReconstruct.py",
+        "ForwardTaskRequest.py",
+        "FreeObjectsRequest.py",
+        "GetTaskReply.py",
+        "MessageType.py",
+        "NotifyUnblocked.py",
+        "PushErrorRequest.py",
+        "RegisterClientReply.py",
+        "RegisterClientRequest.py",
+        "RegisterNodeManagerRequest.py",
+        "ResourceIdSetInfo.py",
+        "SubmitTaskRequest.py",
+        "Task.py",
+        "TaskExecutionSpecification.py",
+        "WaitReply.py",
+        "WaitRequest.py"
+    ],
+    out_prefix = "python/ray/core/generated/ray/protocol/",
+    include_paths = [
+        "src/ray/gcs/format/"
+    ],
+    includes = ["src/ray/gcs/format/gcs.fbs"]
+)
+
+filegroup(
+    name = "python_sources",
+    srcs = glob([
+        "python/ray/*.py",
+        "python/ray/autoscaler/*.py",
+        "python/ray/cloudpickle/*.py",
+        "python/ray/core/__init__.py",
+        "python/ray/core/generated/__init__.py",
+        "python/ray/core/generated/ray/__init__.py",
+        "python/ray/core/generated/ray/protocol/__init__.py",
+        "python/ray/experimental/*.py",
+        "python/ray/internal/*.py",
+        "python/ray/WebUI.ipynb",
+        "python/ray/workers/default_worker.py"
+    ])
+)
+
+genrule(
+    name = "redis",
+    cmd = "set -x && curl -sL \"https://github.com/antirez/redis/archive/5.0.3.tar.gz\" | tar xz --strip-components=1 -C . && make && find . && mv ./src/redis-server $(location redis-server)",
+    outs = ["redis-server"]
+)
+
+cc_library(
+    name = "ray_redis_module",
+    hdrs = [
+        "src/ray/gcs/redis_module/redismodule.h",
+        "src/ray/gcs/redis_module/redis_string.h"
+    ],
+    srcs = [
+        "src/ray/gcs/redis_module/ray_redis_module.cc",
+        "src/ray/util/logging.cc"
+    ],
+    deps = [
+      ":ray_common"
+    ]
+)
+
+genrule(
+    name = "pkg",
+    srcs = [
+        "python/ray/_raylet.so",
+        "//:python_sources",
+        "//:python_gcs_fbs_outputs",
+        "//:python_node_manager_fbs_outputs",
+        "//:redis-server",
+        "//:ray_redis_module",
+        "//:raylet",
+        "//:raylet_monitor",
+        "@plasma//:plasma_store_server"
+    ],
+    outs = ["ray_pkg"],
+    cmd = """
+        set -x &&
+        mkdir -p python/ray &&
+        cp $(locations python/ray/_raylet.so) python/ray &&
```

Comment:
What does `locations` do?

Reply:
locations is like location but for multiple locations. Here there is only a single location I think, I changed it.